modules/ephemeral-storage-csi-overview-admin-plugin.adoc

```diff
@@ -0,0 +1,117 @@
+
+.Pod security profile enforcement
+[cols=",^v,^v,^v,^v width="100%",options="header"]
+|===
+|Pod security profile|Driver label: restricted| Driver label: baseline | Driver label: privileged
+
+|Restricted
+|Allowed
+|Denied
+|Denied
+
+|Baseline
+|Allowed
+|Allowed
+|Denied
+
+|Privileged
+|Allowed
+|Allowed
+|Allowed
+|===
+
+[id="security-profile-warning"]
+== Pod security profile warning
+The CSI Volume Admission plugin can warn you if the CSI driver’s effective profile is more permissive than the pod security warning profile for the pod namespace. The following table shows when a warning occurs for different pod security profiles for given label values.
+
+.Pod security profile warning
+[cols=",^v,^v,^v,^v width="100%",options="header"]
+|===
+|Pod security profile|Driver label: restricted| Driver label: baseline | Driver label: privileged
+
+|Restricted
+|No warning
+|Warning
+|Warning
+
+|Baseline
+|No warning
+|No warning
+|Warning
+
+|Privileged
+|No warning
+|No warning
+|No warning
+|===
+
+[id="security-profile-audit"]
+== Pod security profile audit
+The CSI Volume Admission plugin can apply audit annotations to the pod if the CSI driver’s effective profile is more permissive than the pod security audit profile for the pod namespace. The following table shows the audit annotation applied for different pod security profiles for given label values.
+
+.Pod security profile audit
+[cols=",^v,^v,^v,^v width="100%",options="header"]
+|===
+|Pod security profile|Driver label: restricted| Driver label: baseline | Driver label: privileged
+
+|Restricted
+|No audit
+|Audit
+|Audit
+
+|Baseline
+|No audit
+|No audit
+|Audit
+
+|Privileged
+|No audit
+|No audit
+|No audit
+|===
+
+[id="admission-plugin-default-behavior"]
+== Default behavior for the CSI Volume Admission plugin
+
+If the referenced CSI driver for a CSI ephemeral volume does not have the `csi-ephemeral-volume-profile` label, the CSI Volume Admission plugin considers the driver to have the privileged profile for enforcement, warning, and audit behaviors. Likewise, if the pod’s namespace does not have the pod security admission label set, the Admission plugin assumes the restricted profile is allowed for enforcement, warning, and audit decisions. Therefore, if no labels are set, CSI ephemeral volumes using that CSI driver are only usable in privileged namespaces by default.
+
+The CSI drivers that ship with {product-title} and support ephemeral volumes have a reasonable default set for the `csi-ephemeral-volume-profile` label:
+
+* Shared Resource CSI driver: restricted
```

Comment:
This is the only technical detail worth preserving - pods in any namespace can use the Shared Resource CSI driver.